tests/test_patchy.py

```diff
@@ -166,6 +166,160 @@ def method(self):
+            def __init__(self):
+                self.base_prop = 'yo'
+
+        class Artist(Person):
+            def __init__(self):
+                super().__init__()
+                self.prop = 'old'
+
+        patchy.patch(Artist.__init__, """\
+            @@ -1,3 +1,3 @@
+             def __init__(self):
+                 super().__init__()
+            -    self.prop = 'old'
+            +    self.prop = 'new'""")
+
+        a = Artist()
+        assert a.base_prop == 'yo'
+        assert a.prop == 'new'
+
+    def test_patch_freevars(self):
+        def free_func(v):
+            return v + ' on toast'
+
+        def sample():
+            filling = 'Chalk'
+            return free_func(filling)
+
+        patchy.patch(sample, """\
+            @@ -1,2 +1,2 @@
+             def method():
+            -    filling = 'Chalk'
+            +    filling = 'Cheese'
+                 return free_func(filling)
+            """)
+
+        assert sample() == "Cheese on toast"
+
+    def test_patch_freevars_order(self):
+        def tastes_good(v):
+            return v + ' tastes good'
+
+        def tastes_bad(v):
+            return v + ' tastes bad'
+
+        def sample():
+            return ', '.join([
+                tastes_good('Cheese'),
+                tastes_bad('Chalk'),
+            ])
+
+        patchy.patch(sample, """\
+            @@ -1,4 +1,4 @@
+             def sample():
+                 return ', '.join([
+            -        tastes_good('Cheese'),
+            -        tastes_bad('Chalk'),
+            +        tastes_bad('Chalk'),
+            +        tastes_good('Cheese'),
+                 )]
+            """)
+
+        assert sample() == 'Chalk tastes bad, Cheese tastes good'
+
+    def test_patch_freevars_remove(self):
+        def tastes_good(v):
+            return v + ' tastes good'
+
+        def tastes_bad(v):
+            return v + ' tastes bad'
+
+        def sample():
+            return ', '.join([
+                tastes_bad('Chalk'),
+                tastes_good('Cheese'),
+            ])
+
+        patchy.patch(sample, """\
+            @@ -1,5 +1,4 @@
+             def sample():
+                 return ', '.join([
+            -        tastes_bad('Chalk'),
+                     tastes_good('Cheese'),
+                 ])
+            """)
+
+        assert sample() == 'Cheese tastes good'
+
+    def test_patch_freevars_nested(self):
+        def free_func(v):
+            return v + ' on toast'
+
+        def sample():
+            filling = 'Chalk'
+
+            def _inner_func():
+                return free_func(filling)
+
+            return _inner_func
+
+        patchy.patch(sample, """\
+            @@ -1,2 +1,2 @@
+             def sample():
+            -    filling = 'Chalk'
+            +    filling = 'Cheese'
+
+                 def _inner_func():
+            """)
+
+        assert sample()() == "Cheese on toast"
+
+    @pytest.mark.xfail(raises=NameError)
```

Comment:
I'm pretty sure it could be done by digging around in the garbage collector. But it would be unreliable